src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test.Common/Xunit/ForegroundTheoryDiscoverer.cs

```diff
@@ -2,24 +2,24 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
-using System.Linq;
 using Xunit.Abstractions;
 using Xunit.Sdk;
 
 namespace Xunit
 {
-    internal class ForegroundTheoryDiscoverer : IXunitTestCaseDiscoverer
+    internal class ForegroundTheoryDiscoverer : TheoryDiscoverer
     {
-        private readonly TheoryDiscoverer _inner;
-
         public ForegroundTheoryDiscoverer(IMessageSink diagnosticMessageSink)
+            : base(diagnosticMessageSink)
         {
-            _inner = new TheoryDiscoverer(diagnosticMessageSink);
         }
 
-        public IEnumerable<IXunitTestCase> Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
+        protected override IEnumerable<IXunitTestCase> CreateTestCasesForDataRow(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo theoryAttribute, object[] dataRow)
```

Comment:
for future lookers - I'm not certain that this is enough - this seems to handle the happy path. We might need to do more for things like skips, or the case where serialization is not supported.